lib/ansible/plugins/action/vyos.py

```diff
@@ -81,9 +81,9 @@ def run(self, tmp=None, task_vars=None):
 
         conn = Connection(socket_path)
         out = conn.get_prompt()
-        while to_text(out, errors='surrogate_then_replace').strip().endswith(')#'):
+        while to_text(out, errors='surrogate_then_replace').strip().endswith('#'):
             display.vvvv('wrong context, sending exit to device', self._play_context.remote_addr)
-            conn.send_command('abort')
+            conn.send_command('exit')
```

Comment:
Since this check-in action plugin ensures the prompt is out of `configure` mode, we can use `exit discard` here as it always ensures the prompt is changed to operational mode.

Reply:
Yup. I wanted one more pair of eyes to approve this approach.